meson.build

```diff
@@ -338,7 +347,7 @@ ok = cc.has_header_symbol('sys/procctl.h', 'PROC_ASLR_CTL')
 userconf.set10('HAVE_DECL_PROCCTL_ASLR_CTL', ok)
 
 if host_machine.system() == 'freebsd'
-  add_project_link_arguments('-Wl,--unresolved-symbols,ignore-in-object-files', language: 'c')
+  add_project_link_arguments('-Wl,--unresolved-symbols,ignore-in-object-files', language: ['c', 'cpp'])
```

Comment:
is this really needed? Why? Maybe add a comment above this line to explain if it's *really* necessary.

Reply:
Yes. it is needed. i spent my sunday on this because the C code was compiling but not the c++ one due environ not being available in the so file which is true, because it is only available when compiling the main.

Reply:
My bad